internal/controllers/reconciliation/patch_test.go

```diff
@@ -299,3 +300,164 @@ func TestPatchDeletionBeforeUpgrade(t *testing.T) {
 	require.NotEqual(t, creationUID, recreationUID)
 	require.Equal(t, val, cm.Data[key])
 }
+
+// TestPatchDeletionForResourceWithReconciliation proves that a patch resource can delete the resource it references
+// even the resource with reconcile-interval annotation and it won't be re-created again.
+func TestPatchDeletionForResourceWithReconciliation(t *testing.T) {
+	scheme := runtime.NewScheme()
+	corev1.SchemeBuilder.AddToScheme(scheme)
+	testv1.SchemeBuilder.AddToScheme(scheme)
+
+	ctx := testutil.NewContext(t)
+	mgr := testutil.NewManager(t)
+	upstream := mgr.GetClient()
+	downstream := mgr.DownstreamClient
+
+	// Test configmap setup
+	cmName := "test-obj"
+	cmNamespace := "default"
+	cm := &corev1.ConfigMap{}
+	cm.Name = cmName
+	cm.Namespace = cmNamespace
+	cm.Annotations = map[string]string{
+		"eno.azure.io/readiness-group":    "2",
+		"eno.azure.io/reconcile-interval": "1ms",
+	}
+
+	registerControllers(t, mgr)
+	testutil.WithFakeExecutor(t, mgr, func(ctx context.Context, s *apiv1.Synthesizer, input *krmv1.ResourceList) (*krmv1.ResourceList, error) {
+		obj := &unstructured.Unstructured{
+			Object: map[string]any{
+				"apiVersion": "eno.azure.io/v1",
+				"kind":       "Patch",
+				"metadata": map[string]any{
+					"name":      cmName,
+					"namespace": cmNamespace,
+					"annotations": map[string]string{
+						// This patch should be applied before the configmap is re-created.
+						"eno.azure.io/readiness-group": "1",
+					},
+				},
+				"patch": map[string]any{
+					"apiVersion": "v1",
+					"kind":       "ConfigMap",
+					"ops": []map[string]any{
+						{"op": "add", "path": "/metadata/deletionTimestamp", "value": "2024-01-22T19:13:15Z"},
+					},
+				},
+			},
+		}
+
+		return &krmv1.ResourceList{Items: []*unstructured.Unstructured{obj}}, nil
+	})
+
+	// Test subject
+	setupTestSubject(t, mgr)
+	mgr.Start(t)
+
+	// Create configmap first before the patch deletion is applied.
+	require.NoError(t, downstream.Create(ctx, cm))
```

Comment:
We should create `cm` with Eno instead of directly with client-go - Eno only "knows about" the reconcile-interval of resources it manages.

Reply:
Actually- maybe we can just remove this test? `TestPatchDeletionForResourceWithReconciliationFromInput` covers the scenario I had in mind

Reply:
i was trying to check the behavior of applying the patch deletion to the existing resource, but you are right that we should test for the resource created by eno only. Remove this test.